critwall/src/main/java/org/opencadc/critwall/BucketSelector.java

```diff
@@ -67,18 +67,71 @@
 
 package org.opencadc.critwall;
 
+import ca.nrc.cadc.util.HexUtil;
+import ca.nrc.cadc.util.StringUtil;
 import org.apache.log4j.Logger;
+import org.opencadc.inventory.InventoryUtil;
 
 /**
  *
  * @author pdowler
  */
 public class BucketSelector {
-    private final Logger log = Logger.getLogger(BucketSelector.class);
-    public final String bucketSelector;
+    private static final Logger log = Logger.getLogger(BucketSelector.class);
+    private final String bucketSelectors;
+    private final int min;
+    private final int max;
+    private final String hexMin;
+    private final String hexMax;
+    private final String hexBuffer = "000";
 
-    BucketSelector(String selectorPrefix) {
-        this.bucketSelector = selectorPrefix;
-        throw new UnsupportedOperationException("TODO");
+    public BucketSelector(String selectors) {
+        InventoryUtil.assertNotNull(BucketSelector.class, "selectors", selectors);
+        this.bucketSelectors = selectors;
+
+        // For first iteration, selectors can only be a range
+        // Check that selector range only uses 0..15
+        String[] minMax = selectors.split("-");
+
+        if (minMax.length > 2) {
+            throw new IllegalArgumentException("invalid bucket selector: single value or range only.");
+        } else {
+            try {
+                hexMin = StringUtil.trimTrailingWhitespace(StringUtil.trimLeadingWhitespace(minMax[0])).toLowerCase();
+                min = HexUtil.toShort(hexBuffer + hexMin);
+                if (minMax.length == 1) {
+                    hexMax = hexMin;
+                    max = min;
+                } else {
+                    hexMax = StringUtil.trimTrailingWhitespace(StringUtil.trimLeadingWhitespace(minMax[1])).toLowerCase();
+                    max = HexUtil.toShort(hexBuffer + hexMax);
+                }
+                log.debug("values: " + hexMin + " " + min);
+            } catch (Exception e) {
+                log.debug("min toShort failed" + e);
+                throw new IllegalArgumentException("invalid format: " + selectors, e);
+            }
+        }
+
+        if (min < 0 || max < min || max > 15) {
+            throw new IllegalArgumentException("invalid bucket selector (min,max): " + min + "," + max);
+        }
+    }
+
```

Comment:
I assume min and max is the API that FileSync will use to get values for calling ArtifactDAO.unstoredIterator(String), so FileSync needs string bucket prefixes.

Why are the int(s) exposed? What are the int(s)? 

Say the buckets value is "a - d"... that means FileSync has to call unstoredIterator(bp) with values bp = a, b, c, d... so some code is going to have to create a sequence of strings with those 4 bucket prefixes. I would put that sequence generation in BucketSelector.

sequence is either a List<String> or an Iterator<String>... since it is code-generated, there isn't much point in creating all the values up front (list)... so Iterator it is. These four method could be replaced by:
```
public Iterator<String> getBuckets()
```
and then you could implement a little of FileSync.run() to see if the approach makes sense. Something like:
```
Iterator<String> buckets = bs.getBuckets();
while (buckets.hasNext()) {
   String b = buckets.next();
   log.info("processing bucket: " + b);
}
throw new UnsupportedOperationException("TODO");
```
(I put the throws after the loop so multiple log messages can be seen when the container is run)

It's always better to get the connectivity between things worked out correctly early on and write at least some of the code on both sides (the API and the use of the API) at the same time.